sig/datadog/core/telemetry/component.rbs

```diff
@@ -11,7 +11,7 @@ module Datadog
 
         include Core::Utils::Forking
 
-        def initialize: (heartbeat_interval_seconds: Float, metrics_aggregation_interval_seconds: Float, dependency_collection: bool, ?enabled: bool, ?metrics_enabled: bool) -> void
+        def initialize: (http_transport: Datadog::Core::Telemetry::Http::Transport, heartbeat_interval_seconds: Float, metrics_aggregation_interval_seconds: Float, dependency_collection: bool, ?enabled: bool, ?metrics_enabled: bool, shutdown_timeout_seconds: Float) -> void
```

Comment:
Minor:

```suggestion
        def initialize: (http_transport: Datadog::Core::Telemetry::Http::Transport, heartbeat_interval_seconds: Float, metrics_aggregation_interval_seconds: Float, dependency_collection: bool, ?enabled: bool, ?metrics_enabled: bool, shutdown_timeout_seconds: Numeric) -> void
```

(To allow integers as well?)

Reply:
I need to see if this works: in Ruby Numeric does not have any arithmetic operators, so if any are used down the line, numeric won't work...

Reply:
Interesting... I guess `Integer | Float` maybe? Although I do admit is a bit of a pain to use.